task/backend/scheduler.go

```diff
@@ -286,7 +269,7 @@ func (s *TickScheduler) Stop() {
 	s.executor.Wait()
 }
 
-func (s *TickScheduler) ClaimTask(task *StoreTask, meta *StoreTaskMeta) (err error) {
+func (s *TickScheduler) ClaimTask(authCtx context.Context, task *platform.Task) (err error) {
```

Comment:
`authCtx` is a good annotation here that we expect the task to have an authorizer.

Should ClaimTask fail early if there is no authorizer on it?

Reply:
Im not actually sure. Auth is required during the transition period because the TaskControlAdaptor makes calls to our storage system. I think validation in the future will probably happen before things get to the scheduler. I think for now we can catch it below the scheduler but It may be arranged differently in the stack in the future.